ovirt/suite_test.go

```diff
@@ -0,0 +1,773 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	ovirtsdk4 "github.com/ovirt/go-ovirt"
+
+	"github.com/ovirt/terraform-provider-ovirt/ovirt"
+)
+
+func getOvirtTestSuite(t *testing.T) OvirtTestSuite {
+	suite, err := NewOvirtTestSuite(
+		os.Getenv("OVIRT_USERNAME"),
+		os.Getenv("OVIRT_PASSWORD"),
+		os.Getenv("OVIRT_URL"),
+		os.Getenv("OVIRT_INSECURE") != "" && os.Getenv("OVIRT_INSECURE") != "0",
+		os.Getenv("OVIRT_CAFILE"),
+		os.Getenv("OVIRT_CA_BUNDLE"),
+		os.Getenv("OVIRT_TEST_CLUSTER_ID"),
+		os.Getenv("OVIRT_BLANK_TEMPLATE_ID"),
+		os.Getenv("OVIRT_TEST_STORAGEDOMAIN_ID"),
+		os.Getenv("OVIRT_TEST_IMAGE_PATH"),
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return suite
+}
+
+func NewOvirtTestSuite(
+	ovirtUsername string,
+	ovirtPassword string,
+	ovirtURL string,
+	ovirtInsecure bool,
+	ovirtCAFile string,
+	ovirtCABundle string,
+	ovirtTestClusterID string,
+	ovirtBlankTemplateID string,
+	testStorageDomainID string,
+	testImageSourcePath string,
+) (suite OvirtTestSuite, err error) {
+	if ovirtUsername == "" {
+		return nil, fmt.Errorf("OVIRT_USERNAME not set for test case")
+	}
+	if ovirtPassword == "" {
+		return nil, fmt.Errorf("OVIRT_PASSWORD not set for test case")
+	}
+	if ovirtURL == "" {
+		return nil, fmt.Errorf("OVIRT_URL not set for test case")
+	}
+	if !ovirtInsecure && ovirtCAFile == "" && ovirtCABundle == "" {
+		return nil, fmt.Errorf("OVIRT_INSECURE, OVIRT_CAFILE, or OVIRT_CA_BUNDLE must be set for acceptance tests")
+	}
+
+	provider := ovirt.Provider().(*schema.Provider)
+	if err := provider.Configure(
+		&terraform.ResourceConfig{
+			Config: map[string]interface{}{
+				"username": ovirtUsername,
+				"password": ovirtPassword,
+				"url": ovirtURL,
+				"insecure": ovirtInsecure,
+				"cafile": ovirtCAFile,
+				"ca_bundle": ovirtCABundle,
+			},
+		},
+	); err != nil {
+		return nil, fmt.Errorf("failed to configure oVirt Terraform provider (%w)", err)
+	}
+	providers := map[string]terraform.ResourceProvider{
+		"ovirt": provider,
+	}
+	conn := provider.Meta().(*ovirtsdk4.Connection)
+
+	if ovirtTestClusterID == "" {
+		ovirtTestClusterID, err = findTestClusterID(conn)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if ovirtBlankTemplateID == "" {
+		ovirtBlankTemplateID, err = findBlankTemplateID(conn)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if testStorageDomainID == "" {
+		testStorageDomainID, err = findStorageDomainID(conn)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	storageDomain, err := findStorageDomain(conn, testStorageDomainID)
+	if err != nil {
+		return nil, err
+	}
+
+	testCluster, err := getTestCluster(conn, ovirtTestClusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	if testImageSourcePath == "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return nil, fmt.Errorf("can't fetch current working directory (%w)", err)
+		}
+		testImageSourcePath = path.Join(cwd, "ovirt", "testimage", "image")
```

Comment:
=== RUN   TestAccOvirtNicsDataSource_nameRegexFilter
    testing.go:654: Step 0 error: errors during apply:
        
        Error: Failed preparing disk for transfer: open /home/eslutsky/GolandProjects/terraform-provider-ovirt/ovirt/ovirt/testimage/image: no such file or directory
        
          on /tmp/tf-test360349721/main.tf line 2:
          (source code not available)
      
it works without  "ovirt" in  path:
`testImageSourcePath = path.Join(cwd, "testimage", "image")`

Reply:
Which folder did you run this from?

Reply:
project root `/home/eslutsky/GolandProjects/terraform-provider-ovirt`

Reply:
Thanks, I'll look into that, it should work from the project root.